runtime/interpreter/interpreter.go

```diff
@@ -2573,11 +2573,188 @@ func init() {
 			panic(fmt.Sprintf("missing converter for number type: %s", numberType))
 		}
 	}
+
+	// We assign this here because it depends on the interpreter, so this breaks the initialization cycle
+	defineBaseValue(
+		baseActivation,
+		"DictionaryType",
+		NewHostFunctionValue(
+			func(invocation Invocation) Value {
+
+				keyType := invocation.Arguments[0].(TypeValue).Type
+
+				// if the given key is not a valid dictionary key, it wouldn't make sense to create this type
+				if keyType == nil ||
+					!sema.IsValidDictionaryKeyType(invocation.Interpreter.ConvertStaticToSemaType(keyType)) {
+					return NilValue{}
+				}
+
+				return NewSomeValueOwningNonCopying(TypeValue{
+					Type: DictionaryStaticType{
+						KeyType:   keyType,
+						ValueType: invocation.Arguments[1].(TypeValue).Type,
+					}})
+			},
+			sema.DictionaryTypeFunctionType,
+		))
+
+	defineBaseValue(
+		baseActivation,
+		"CompositeType",
+		NewHostFunctionValue(
+			func(invocation Invocation) Value {
+				typeID := invocation.Arguments[0].(*StringValue).Str
+				location, qualifiedIdentifier, err := common.DecodeTypeID(typeID)
+
+				// if the typeID is invalid, return nil
+				if err != nil || location == nil {
+					return NilValue{}
+				}
+
+				composite, err := invocation.Interpreter.getCompositeType(location, qualifiedIdentifier)
+				if err != nil {
+					return NilValue{}
+				}
+
+				return NewSomeValueOwningNonCopying(TypeValue{
+					Type: ConvertSemaToStaticType(composite),
+				})
+			},
+			sema.CompositeTypeFunctionType,
+		),
+	)
+
+	defineBaseValue(
+		baseActivation,
+		"InterfaceType",
+		NewHostFunctionValue(
+			func(invocation Invocation) Value {
+				typeID := invocation.Arguments[0].(*StringValue).Str
+				location, qualifiedIdentifier, err := common.DecodeTypeID(typeID)
+
+				// if the typeID is invalid, return nil
+				if err != nil || location == nil {
```

Comment:
Same here, maybe remove the location check

Reply:
The location check is safe to remove for composite types, but not for interfaces; the implementation of `getInterfaceType` requires the location to be non-`nil`: https://github.com/onflow/cadence/blob/master/runtime/interpreter/interpreter.go#L3462

Reply:
Good catch! 👍